stealth.lua

```diff
@@ -0,0 +1,107 @@
+            return err, msg
+        else
+            return privs_command(caller, param)
+        end
+    end
+})
+
+minetest.override_chatcommand("mods", {
+    func = function(caller, param)
+        local modnames = minetest.get_modnames()
+
+        if not (minetest.check_player_privs(caller, "cloaking") or minetest.check_player_privs(caller, "admin")) then
+            local i
+            for idx, modname in ipairs(modnames) do
+                if modname == "cloaking" then
+                    i = idx
+                    break
+                end
+            end
+
+            if i then
+                table.remove(modnames, i)
+            end
+        end
+
+        return true, table.concat(modnames, ", ")
+    end
+})
+
+local function format_help_line(cmd, def)
+    local msg = core.colorize("#00ffff", "/" .. cmd)
+    if def.params and def.params ~= "" then
+        msg = msg .. " " .. def.params
+    end
+    if def.description and def.description ~= "" then
+        msg = msg .. ": " .. def.description
+    end
+    return msg
+end
+
+local help_command = minetest.registered_chatcommands["help"].func
+minetest.override_chatcommand("help", {
+    func = function(name, param)
+        if minetest.check_player_privs(name, "cloaking") or minetest.check_player_privs(name, "admin") then
+            return help_command(name, param)
+        elseif param == "privs" then
+            local privs = {}
+            for priv, def in pairs(minetest.registered_privileges) do
+                if priv ~= "cloaking" then
+                    privs[#privs + 1] = priv .. ": " .. def.description
+                end
+            end
+            table.sort(privs)
+            return true, "Available privileges:\n"..table.concat(privs, "\n")
+        elseif param == "all" then
+            local cmds = {}
+            for cmd, def in pairs(minetest.registered_chatcommands) do
+                if minetest.check_player_privs(name, def.privs) and cmd ~= "uncloak" then
+                    cmds[#cmds + 1] = format_help_line(cmd, def)
+                end
+            end
+            table.sort(cmds)
+            return true, "Available commands:".."\n"..table.concat(cmds, "\n")
+        elseif param == "" then
+            local cmds = {}
+            for cmd, def in pairs(minetest.registered_chatcommands) do
+                    if core.check_player_privs(name, def.privs) and cmd ~= "uncloak" then
+                        cmds[#cmds + 1] = cmd
+                    end
+            end
+            table.sort(cmds)
+            return true, "Available commands: " .. table.concat(cmds, " ") .. "\n"
+                            .. "Use '/help <cmd>' to get more information,"
+                            .. " or '/help all' to list everything."
+        elseif param == "cloak" or param == "uncloak" then
+            return false, "Command not available: "..param
+
+        else
+            return help_command(name, param)
+        end
+    end
+})
```

Comment:
Is anything nicer than recreating `/help` possible?

Reply:
There's certainly other options, but this felt the cleanest to me. I certainly feel it's not ideal, as /help implementations might change at any point, but then again, so might the rest of the game. Outside of overriding help (and similar things) the best option I can think of is effectively creating an entire second "stealth" privilege system, which might be more correct but also much more complicated to get right.